airbyte-connector-builder-server/connector_builder/impl/default_api.py

```diff
@@ -164,33 +177,53 @@ def _get_message_groups(self, messages: Iterator[AirbyteMessage], schema_inferre
 
         Note: The exception is that normal log messages can be received at any time which are not incorporated into grouping
         """
-        first_page = True
-        current_records = []
+        records_count = 0
+        at_least_one_page_in_group = False
+        current_page_records = []
+        current_slice_pages = []
         current_page_request: Optional[HttpRequest] = None
         current_page_response: Optional[HttpResponse] = None
 
-        while len(current_records) < limit and (message := next(messages, None)):
-            if first_page and message.type == Type.LOG and message.log.message.startswith("request:"):
-                first_page = False
-                request = self._create_request_from_log_message(message.log)
-                current_page_request = request
+        while records_count < limit and (message := next(messages, None)):
+            if self._need_to_close_page(at_least_one_page_in_group, message):
+                self._close_page(current_page_request, current_page_response, current_slice_pages, current_page_records)
```

Comment:
should this branch also set `current_page_request` and `current_page_response` to `None` to protect ourselves against a malformed stream of messages (due to a bug in the CDK)?

example scenario:
1. request
2. response
3. record
4. (no request)
5. response